frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/activity-card/activity-card.tsx

```diff
@@ -5,66 +5,99 @@ import DashboardCard from '@console/shared/src/components/dashboard/dashboard-ca
-import { CEPH_STORAGE_NAMESPACE, OCS_OPERATOR } from '../../../../constants/index';
+import { SubscriptionKind } from '@console/operator-lifecycle-manager';
+import {
+  CEPH_STORAGE_NAMESPACE,
+  OCS_OPERATOR,
+  PVC_PROVISIONER_ANNOTATION,
+} from '../../../../constants/index';
 import { DATA_RESILIENCY_QUERY, StorageDashboardQuery } from '../../../../constants/queries';
+import {
+  pvcResource,
+  subscriptionResource,
+  storageClusterResource,
+  eventsResource,
+} from '../../../../constants/resources';
 import { getResiliencyProgress } from '../../../../utils';
-import { OCSServiceModel } from '../../../../models';
 import { isClusterExpandActivity, ClusterExpandActivity } from './cluster-expand-activity';
 import { isOCSUpgradeActivity, OCSUpgradeActivity } from './ocs-upgrade-activity';
 import './activity-card.scss';
 
-const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
-const subscriptionResource: FirehoseResource = {
-  isList: true,
-  kind: referenceForModel(SubscriptionModel),
-  namespaced: false,
-  prop: 'subs',
-};
-
-const storageClusterResource: FirehoseResource = {
-  isList: true,
-  kind: referenceForModel(OCSServiceModel),
-  namespaced: false,
-  prop: 'storage-cluster',
-};
-
 export const getOCSSubscription = (subscriptions: FirehoseResult): SubscriptionKind => {
   const itemsData: K8sResourceKind[] = subscriptions?.data;
   return _.find(itemsData, (item) => item?.spec?.name === OCS_OPERATOR) as SubscriptionKind;
 };
 
-const ocsEventNamespaceKindFilter = (event: EventKind): boolean => {
-  const eventKind = event?.involvedObject?.kind;
-  const eventNamespace = getNamespace(event);
-  return eventNamespace === CEPH_STORAGE_NAMESPACE || eventKind === PersistentVolumeClaimModel.kind;
-};
-
 const RecentEvent = withDashboardResources(
   ({ watchK8sResource, stopWatchK8sResource, resources }: DashboardItemProps) => {
     React.useEffect(() => {
       watchK8sResource(eventsResource);
+      watchK8sResource(pvcResource);
       return () => {
         stopWatchK8sResource(eventsResource);
+        stopWatchK8sResource(pvcResource);
       };
     }, [watchK8sResource, stopWatchK8sResource]);
+
+    const validPVC = React.useMemo((): string[] => {
+      const pvcList = resources.pvcs
+        ? (resources?.pvcs?.data as PersistentVolumeClaimKind[]).filter((obj) => {
+            const annotations = getAnnotations(obj);
+            return (
+              annotations &&
+              annotations[PVC_PROVISIONER_ANNOTATION] &&
+              isCephProvisioner(annotations[PVC_PROVISIONER_ANNOTATION])
+            );
+          })
+        : [];
+      const pvcNames = _.reduce(
+        pvcList,
+        (res, pvcObj) => {
+          const name = getName(pvcObj);
+          res.push(name);
+          return res;
+        },
+        [],
+      );
+      return pvcNames;
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
```suggestion
  return pvcList?.map(getName);
```
I also don't see the need to disable the eslint rule.